addons/binding/org.openhab.binding.evohome/src/main/java/org/openhab/binding/evohome/internal/api/EvohomeApiClientV2.java

```diff
@@ -95,113 +122,105 @@ private LocationsStatus requestLocationsStatus() {
 
     @Override
     public boolean login() {
-        boolean success = false;
-        try {
-            // Building the HTTP request discretely here, as it is the only one with a deviant content type
-            Request request = httpClient.newRequest(EvohomeApiConstants.URL_V2_AUTH);
-            request.method(HttpMethod.POST);
-            request.header("Authorization", "Basic YjAxM2FhMjYtOTcyNC00ZGJkLTg4OTctMDQ4YjlhYWRhMjQ5OnRlc3Q=");
-            request.header("Accept", "application/json, application/xml, text/json, text/x-json, text/javascript, text/xml");
-
-            String data = "Username=" + URLEncoder.encode(configuration.username, "UTF-8") + "&"
-                    + "Password=" + URLEncoder.encode(configuration.password, "UTF-8") + "&"
-                    + "Host=rs.alarmnet.com%2F&"
-                    + "Pragma=no-cache&"
-                    + "Cache-Control=no-store+no-cache&"
-                    + "scope=EMEA-V1-Basic+EMEA-V1-Anonymous+EMEA-V1-Get-Current-User-Account&"
-                    + "grant_type=password&"
-                    + "Content-Type=application%2Fx-www-form-urlencoded%3B+charset%3Dutf-8&"
-                    + "Connection=Keep-Alive";
-
-            request.content(new StringContentProvider(data), "application/x-www-form-urlencoded");
-
-            ContentResponse response = request.send();
-            if (response.getStatus() == 200) {
-                String reply = response.getContentAsString();
-                success = true;
-                apiAccess.setAuthentication(new Gson().fromJson(reply, Authentication.class));
-            }
-        } catch (Exception e) {
-            apiAccess.setAuthentication(null);
-            logger.error("Authorization failed", e);
-        }
+       boolean success = authenticate();
 
         // If the authentication succeeded, gather the basic intel as well
         if (success == true) {
-            useraccount = requestUserAccount();
-            locations   = requestLocations();
+            useraccount        = requestUserAccount();
+            locations          = requestLocations();
```

Comment:
By the way, I was running into the same issue you had with the null reference. In `EvohomeApiClientV2.getHeatingZone()` you are calling `getLocationStatus()`. That one is not guaranteed to have valid info. I think it is also not the call you are interested in, since it only contains actual values and a little bit metadata
So This

              {
                "zoneId": "[id of zone]",
                "modelType": "HeatingZone",
                "heatSetpointCapabilities": {
                  "maxHeatSetpoint": 35.0000,
                  "minHeatSetpoint": 5.0000,
                  "valueResolution": 0.5,
                  "allowedSetpointModes": [
                    "PermanentOverride",
                    "FollowSchedule",
                    "TemporaryOverride"
                  ],
                  "maxDuration": "1.00:00:00",
                  "timingResolution": "00:10:00"
                },
                "scheduleCapabilities": {
                  "maxSwitchpointsPerDay": 6,
                  "minSwitchpointsPerDay": 1,
                  "timingResolution": "00:10:00",
                  "setpointValueResolution": 0.5
                },
                "name": "[zone name]",
                "zoneType": "RadiatorZone"
              }

Versus this

            {
              "zoneId": "[zone id]",
              "temperatureStatus": {
                "temperature": 24.5,
                "isAvailable": true
              },
              "activeFaults": [],
              "heatSetpointStatus": {
                "targetTemperature": 15.0000,
                "setpointMode": "FollowSchedule"
              },
              "name": "[zone name]"
            }

The first also gives you info on what the range of temperature setpoint is to create the slider dynamically.